src/Common/src/Interop/Windows/advapi32/Interop.CryptSignHashW.cs

```diff
@@ -0,0 +1,48 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Runtime.InteropServices;
+using System.Security.Cryptography;
+
+internal partial class Interop
+{
+    internal partial class Advapi32
+    {
+        internal enum KeySpec : int
+        {
+            AT_KEYEXCHANGE = 1,
+            AT_SIGNATURE = 2,
+        }
+
+        [Flags]
+        internal enum CryptSignAndVerifyHashFlags : int
+        {
+            None = 0x00000000,
+            CRYPT_NOHASHOID = 0x00000001,
+            CRYPT_TYPE2_FORMAT = 0x00000002,  // Not supported
+            CRYPT_X931_FORMAT = 0x00000004,  // Not supported
+        }
+
+        [DllImport(Libraries.Advapi32, CharSet = CharSet.Unicode, SetLastError = true, EntryPoint = "CryptSignHashW")]
+        [return: MarshalAs(UnmanagedType.Bool)]
+        public static extern bool CryptSignHash(
+            SafeHashHandle hHash,
+            KeySpec dwKeySpec,
+            string szDescription,
+            CryptSignAndVerifyHashFlags dwFlags,
+            [Out] byte[] pbSignature,
+            [In, Out] ref int pdwSigLen);
+
+        [DllImport(Libraries.Advapi32, CharSet = CharSet.Unicode, SetLastError = true, EntryPoint = "CryptVerifySignatureW")]
+        [return: MarshalAs(UnmanagedType.Bool)]
+        public static extern bool CryptVerifySignature(
```

Comment:
For these two we have the shared enum  `CryptSignAndVerifyHashFlags`. What would be better: move to the same file CryptSetProvParam and CryptGetProvParam or to separated files Sign and Verify?

Reply:
Personally, I'd put CryptSetProvParam back in the file with CryptGetProvParam and their shared enum.

Reply:
It is ok, but what about `SafeProvHandle`  type? The `SafeProvHandle` is declared in `src/System.Security.Cryptography.Csp/src/System/Security/Cryptography/SafeCryptoHandles.cs` and it is not ok if we will have a reference from common `Interop` to System.Security.Cryptography.Csp, because it also used in other projects. 

Could you please suggest? What about moving  `SafeProvHandle`  and other (SafeKeyHandle, SafeHashHandle) from `src/System.Security.Cryptography.Csp/src/System/Security/Cryptography/SafeCryptoHandles.cs`  to `src/Common/src/Microsoft/Win32/SafeHandles/SafeCryptoHandles.cs`